src/jokebase.rs

```diff
@@ -65,6 +67,15 @@ impl JokeBase {
         self.write_jokes()?;
         Ok(())
     }
+
+    pub fn delete(&mut self, given_id: String) -> Result<(), JokeBaseError> {
+        if self.jokemap.contains_key(&given_id) == false {
+            return Err(JokeBaseError::JokeDoesNotExists(given_id));
+        } else {
+            self.jokemap.remove(&given_id);
+            Ok(())
```

Comment:
```suggestion
            self.jokemap.remove(&given_id);
            self.write_jokes()?;
            Ok(())
```

After updating the jokemap, we should update the on-disk version of the database.
This way, if a crash occurs after the deletion, we’ll be able to restart the database without losing any operations.